test/Microsoft.ApplicationInsights.AspNetCore.Tests/Extensions/HttpRequestExtensionsTest.cs

```diff
@@ -99,7 +100,7 @@ public void TestGetUriReturnsCorrectUriIfRequestObjectSchemeAndHostAndPathAndQue
             var uri = HttpRequestExtensions.GetUri(request);
 
             Assert.Equal(
-                new Uri(string.Format("{0}://{1}{2}{3}", ExpectedSchema, ExpectedHostName, ExpectedPath, ExpectedQueryString)),
+                new Uri(string.Format(CultureInfo.InvariantCulture, "{0}://{1}{2}{3}", ExpectedSchema, ExpectedHostName, ExpectedPath, ExpectedQueryString)),
```

Comment:
The schema and hostname might be safe with Invariant, although I doubt that hostname will always be safe...
However, I'm not sure that the path and query name are bound to punycode rules or anything else that will force them to be ASCII-like.  I don't recall for certain if it could be an issue with URLs or not but it warrants checking.